### PR TITLE
Use explicit linker generated high/low symbols to determine stack bounds. NFC

### PR DESCRIPTION
**Pull request by sbc100**
See https://reviews.llvm.org/D135910

### CODE REVIEW

**Comment by sbc100** [comment]
Depends on https://reviews.llvm.org/differential/revision/edit/135910/

File: emscripten.py [hunk 1]
```diff
@@ -353,7 +353,7 @@ def emscript(in_wasm, out_wasm, outfile_js, memfile):
     dylink_sec = webassembly.parse_dylink_section(in_wasm)
     static_bump = align_memory(dylink_sec.mem_size)
     set_memory(static_bump)
-    logger.debug('stack_base: %d, stack_max: %d, heap_base: %d', settings.STACK_BASE, settings.STACK_MAX, settings.HEAP_BASE)
+    logger.debug('stack_low: %d, stack_high: %d, heap_base: %d', settings.STACK_LOW, settings.STACK_LOW, settings.HEAP_BASE)
```
Comment (kripken) [inline emscripten.py:356]:
```suggestion
    logger.debug('stack_low: %d, stack_high: %d, heap_base: %d', settings.STACK_HIGH, settings.STACK_LOW, settings.HEAP_BASE)
```

File: src/library.js [hunk 2]
```diff
@@ -3617,6 +3617,8 @@ mergeInto(LibraryManager.library, {
   // have __heap_base hardcoded into it - it receives it from JS as an extern
   // global, basically).
   __heap_base: '{{{ HEAP_BASE }}}',
```
Comment (kripken) [inline src/library.js:3619]: do we still need `heap_base` here? (it is removed from `stack_limits.s` but I guess there's another use?)

**Review by kripken — APPROVED** [review]

File: src/library.js (hunk 2 above)
Comment (sbc100) [reply src/library.js:3619]: Yes, this is actually only used in the dynamic linking case.   If you looks this block is inside `#if RELOCATABLE`.  This is a symbol that wasm-ld normally provides but in the case of dynamic linking its expected that the runtime provides it.  

It is still used in emscripten, e.g. in sbrk.c

**Pull request merged**